src/lib/krb5/ccache/cc_kcm.c

```diff
@@ -972,8 +972,10 @@ kcm_start_seq_get(krb5_context context, krb5_ccache cache,
     if (ret == 0) {
         /* GET_CRED_LIST is available. */
         ret = kcmreq_get_cred_list(&req, &creds);
-        if (ret)
+        if (ret) {
+            free_cred_list(creds);
             goto cleanup;
+        }
```

Comment:
This is not a leak; kcmreq_get_cred_list() only allocates creds if it returns successfully.  kcmreq_get_cred_list() attempts to make this clear to static analysis tools by setting *creds_out to NULL on entry and only setting it to a non-null value just before returning 0.

Reply:
I think I misunderstood this case. The leak appends if [this `k5alloc()` call](https://github.com/krb5/krb5/blob/krb5-1.21-final/src/lib/krb5/ccache/cc_kcm.c#L991) fails, because `creds` is not assigned to `cursor` yet.

Reply:
I agree that there is a leak in that case.  uuids also leaks if it was allocated in the `else if` block.